tianshou/policy/modelfree/pg.py

```diff
@@ -103,7 +108,13 @@ def forward(
             dist = self.dist_fn(*logits)
         else:
             dist = self.dist_fn(logits)
-        act = dist.sample()
+        if self._deterministic_eval and not self.training:
+            if isinstance(logits, tuple):
+                act = logits[0].argmax(-1)
```

Comment:
Currently tuple means `(mu, sigma)` for continuous action space, and it seems like using `act = logits[0]` is better.

Another thing: we should check the action space's type in this case, discrete or continuous.

Reply:
Maybe we can check whether dist_fn is `torch.distributions.Categorical`. If it is, we apply an extra argmax?

Does it make sense?

Reply:
> Maybe we can check whether dist_fn is `torch.distributions.Categorical`.

That's not a universal approach. If someone wants to use another dist_fn for discrete action space and `deterministic_eval=True`, it won't result in the expected behavior.

Reply:
Yes. But without argmax, the action shape (action space) would be different in training and evaluation. What's your suggestion?

Reply:
> Can we enforce user to implement

Better not. It would be much difficult when using DiagGaussian distribution.

> What's your suggestion?

Well, my current thought is to check the type of action_space in `BasePolicy.__init__`, and use it here.

Reply:
Make sense to me. Implementation updated.